samples/bluetooth/peripheral_hr/src/main.c

```diff
@@ -32,18 +33,28 @@ static const struct bt_data ad[] = {
 		      BT_UUID_16_ENCODE(BT_UUID_DIS_VAL))
 };
 
+/* Use atomic variable, 2 bits for connection and disconnection state */
+static ATOMIC_DEFINE(state, 2U);
+
+#define STATE_CONNECTED    1U
+#define STATE_DISCONNECTED 2U
```

Comment:
What is the advantage of this over e.g. a `volatile bool`?

Reply:
As callback and the main threads are at different priority, use of atomic flags is context safe compared to use of `volatile bool`